indexer/src/handler.rs

```diff
@@ -106,23 +112,104 @@ impl MessageHandler {
             .context("Transaction not found")?
             .message
             .context("Message not found")?;
+        let sig = tx
+            .transaction
+            .as_ref()
+            .ok_or_else(|| anyhow!("failed to get transaction"))?
+            .signature
+            .clone();
 
-        let mut i = 0;
         let keys = message.clone().account_keys;
 
         for (idx, key) in message.clone().account_keys.iter().enumerate() {
-            let k = Pubkey::new(key);
+            let key: &[u8] = key;
+            let k = Pubkey::try_from(key)?;
             if k == spl_token::ID {
-                i = idx;
-                break;
+                self.process_spl_token_transaction(idx, &keys, &sig, &message)
+                    .await?;
+            } else if k == mpl_bubblegum::ID {
+                self.process_mpl_bubblegum_transaction(idx, &keys, &sig, &message)
+                    .await?;
             }
         }
 
+        Ok(())
+    }
+
+    async fn process_mpl_bubblegum_transaction(
+        &self,
+        program_account_index: usize,
+        keys: &[Vec<u8>],
+        sig: &Vec<u8>,
+        message: &Message,
+    ) -> Result<()> {
         for ins in message.instructions.iter() {
```

Comment:
We aren't checking instructions inner_instructions so aren't accounting for transfers that occur from cpi call like when NFT sold through a marketplace.